plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/engine/AbstractHttpAIClient.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.model.ai.engine;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonSyntaxException;
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.Log;
+import org.jkiss.dbeaver.model.ai.utils.MonitoredHttpClient;
+import org.jkiss.dbeaver.model.data.json.JSONUtils;
+
+import java.net.http.HttpClient;
+
+public abstract class AbstractHttpAIClient implements AutoCloseable {
+    private static final Log log = Log.getLog(AbstractHttpAIClient.class);
+    protected final MonitoredHttpClient client;
+
+    public AbstractHttpAIClient() {
+        this.client = new MonitoredHttpClient(
+            HttpClient.newHttpClient(),
+            this::mapHttpError
+        );
+    }
+
+    @Override
+    public void close() {
+        client.close();
+    }
+
+    @NotNull
+    protected DBException mapHttpError(int statusCode, @NotNull String body) {
+        return new DBException("AI request failed: " + statusCode + ", " + parseErrorMessage(body));
```

Comment:
I  don't like that the message includes both the status code and the prefix "AI request failed". This can be logged, but the exception itself must only contain the "mapped" message. Alternatively, use a separate exception type that includes the "mapped" message as a separate property, which can be retrieved and displayed in the chat.